pkg/reconcilermanager/controllers/reposync_controller.go

```diff
@@ -748,6 +754,27 @@ func (r *RepoSyncReconciler) validateNamespaceName(namespaceName string) error {
 	return nil
 }
 
+func (r *RepoSyncReconciler) validateCACertSecret(ctx context.Context, repoSync *v1beta1.RepoSync) error {
+	caCertSecretRefName := v1beta1.GetSecretName(repoSync.Spec.Git.CACertSecretRef)
+
+	if useCACert(caCertSecretRefName) {
+		secret, err := validateSecretExist(ctx,
+			caCertSecretRefName,
+			repoSync.Namespace,
+			r.client)
+		if err != nil {
+			if apierrors.IsNotFound(err) {
+				return errors.Errorf("Secret %s not found", caCertSecretRefName)
```

Comment:
nit: use a more descriptive error message, consistent with the other Secret error. e.g.
`Secret %s not found, create one to allow client connections with CA certificate`

Reply:
updated error message to be more descriptive